ArduCopter/mode_auto.cpp

```diff
@@ -1072,13 +1138,19 @@ void Copter::ModeAuto::do_nav_wp(const AP_Mission::Mission_Command& cmd)
         }
     }
 
+    if(target_loc.alt!=current_loc.alt){
+        target_loc.alt = current_loc.alt;
```

Comment:
This isn't mentioned in the parameter description, it changes the altitude of the current waypoint target to be whatever altitude the pilot is controlling it to get to.
This doesn't work with splines (or other navigation commands).

Reply:
revised and changed